include/highfive/bits/H5Dataspace_misc.hpp

```diff
@@ -206,6 +206,18 @@ DataSpace::From(const std::vector<Eigen::Matrix<Value, M, N>>& vec) {
     return DataSpace(dims);
 }
 
+#ifdef H5_USE_BOOST
+template <typename Value, int M, int N, size_t Dims>
+inline DataSpace DataSpace::From(const boost::multi_array<Eigen::Matrix<Value, M, N>, Dims>& vec) {
+    std::vector<size_t> dims(Dims);
```

Comment:
Using this `vector` constructor prevents the `for` loop
```
std::vector<size_t> dims(vec.shape(), vec.shape() + Dims)
```

Reply:
Indeed, also updated another part in the code with this.